theforeman.org/scripts/debian/setup_sources_core.sh

```diff
@@ -0,0 +1,83 @@
+#!/bin/bash
+set -xe
+
+# This script clones the git repos, cleans them up, and builds the package
+echo "--Setting Up Sources"
+
+# Setup the debian files, figure out the version
+echo `git log -n1 --oneline`
+cd debian/${os}/
+VERSION=$(head -n1 ${project}/changelog|awk '{print $2}'|sed 's/(//;s/)//'|cut -f1 -d-)
+
+# Setup sources
+mkdir build-${project} && cd build-${project}
+if [[ x$repo =~ ^xdevelop ]]; then
+  url_base='https://ci.theforeman.org'
+  job_name=$nightly_jenkins_job  # e.g. 'test_develop', from triggering job
+  job_id=$nightly_jenkins_job_id # e.g. '123', from triggering
+  json_url="${url_base}/job/${job_name}/${job_id}/api/json"
+
+  # If a last* alias was used, resolve the numeric job ID
+  job_id=$(curl "${json_url}" | /usr/local/bin/JSON.sh -b | awk '$1 == "[\"number\"]" { print $2 }')
+  base_url=`curl "${json_url}" | /usr/local/bin/JSON.sh -b | awk '$1 ~ /^\["runs",.*,"number"\]/ && $2 == '$job_id' {getline; print $2; exit}' | tr -d \"`
+  if [ x$base_url = x ] ; then
+    base_url=`curl "${json_url}" | /usr/local/bin/JSON.sh -b | egrep '\["url"\]' | awk '{print $NF}' | tr -d \"`
+  fi
+  url="${base_url}/artifact/*zip*/archive.zip"
+
+  wget $url
+  unzip archive.zip
+  mv archive/pkg/*bz2 ${project}_${VERSION}.orig.tar.bz2
+
+  # Set this in case we need it
+  LAST_COMMIT=`curl "${json_url}" | /usr/local/bin/JSON.sh -b | egrep '"lastBuiltRevision","SHA1"' | awk '{print $NF}' | tr -d \" | head -n1`
+else
+  VERSION=`echo ${VERSION} | tr '~rc' '-rc'`
+  # Download sources
+  wget https://downloads.theforeman.org/${project}/${project}-${VERSION}.tar.bz2 https://downloads.theforeman.org/${project}/${project}-${VERSION}.tar.bz2.sig
+
+  # Verify with packaging key - commented until we can handle multiple keys
+#  tmp_keyring="./1AA043B8-keyring.gpg"
+#  gpg --no-default-keyring --keyserver keys.gnupg.net --keyring $tmp_keyring --recv-keys 1AA043B8
+#  if gpg --no-default-keyring --keyring $tmp_keyring ${project}-${VERSION}.tar.bz2.sig 2>&1 | grep -q "gpg: Good signature from \"Foreman Automatic Signing Key (2014) <packages@theforeman.org>\"" ; then
+#    true # ok
+#  else
+#    exit 2
+#  fi
```

Comment:
```suggestion
```

unused code can go for a walk

Reply:
Can't we fix the TODO? Isn't that basically add an additional key to the keyring and grep for `gpg: Good signature from`?

Reply:
Grepping gpg output sounds awful, gpg has `--verify`.

But I think we should add that to the TODO list for the re-write, not try to fix it now, as it has been disabled for a while now.

Reply:
I took a stab at it and I think it should be something like this
```sh
GPG_KEYSERVER=pgp.mit.edu
GPG_KEYS="1AA043B8 C8B9E30B0DBCAB595E98DF5C965467189B776908"

# Download sources
wget https://downloads.theforeman.org/${project}/${project}-${VERSION}.tar.bz2 https://downloads.theforeman.org/${project}/${project}-${VERSION}.tar.bz2.sig

# Verify with packaging key
tmp_keyring="$(mktemp)"
gpg2 --no-default-keyring --keyserver $GPG_KEYSERVER --keyring $tmp_keyring --recv-keys $GPG_KEYS
if gpg2 --no-default-keyring --keyring $tmp_keyring --verify ${project}-${VERSION}.tar.bz2.sig ; then
  rm $tmp_keyring
  true # ok
else
  rm $tmp_keyring
  exit 2
fi
```

And I did have some issues with pgp.mit.edu, but that's where we upload it:
https://github.com/theforeman/theforeman-rel-eng/blob/7d8f8c69f935c7912088dd5137520ddeb165d219/upload_gpg#L15

And yes, we should also specify the long key for the old key since short keys are vulnerable.

Reply:
As I do not want to pile on instability in this PR since it already is changing structure I have captured this as an issue against the milestone: https://github.com/theforeman/jenkins-jobs/issues/91